.github/actions/deploy-macosx/action.yml

```diff
@@ -34,25 +34,27 @@ runs:
-          sudo install_name_tool -change $(otool -L /usr/local/lib/gcc/10/libgfortran.5.dylib | grep /usr/local/Cellar/gcc@10/ | cut -f1 -d ' ') @rpath/libquadmath.0.dylib /usr/local/lib/gcc/10/libgfortran.5.dylib
-          sudo install_name_tool -change /usr/local/lib/gcc/10/libgcc_s.1.dylib @rpath/libgcc_s.1.dylib /usr/local/lib/gcc/10/libgomp.1.dylib
-          sudo install_name_tool -change /usr/local/lib/gcc/10/libgcc_s.1.dylib @rpath/libgcc_s.1.dylib /usr/local/lib/gcc/10/libstdc++.6.dylib
-          sudo install_name_tool -change /usr/local/lib/gcc/10/libgcc_s.1.dylib @rpath/libgcc_s.1.dylib /usr/local/lib/gcc/10/libgfortran.5.dylib
-          sudo install_name_tool -change /usr/local/lib/gcc/10/libgcc_s.1.dylib @rpath/libgcc_s.1.dylib /usr/local/lib/gcc/10/libquadmath.0.dylib
-          otool -L /usr/local/lib/gcc/10/*.dylib
+          if [ $(arch) = "arm64" ]; then
+            brew install gcc@13
+            brew link --overwrite gcc@13
+          else
+            brew install gcc@10
+            brew link --overwrite gcc@10
+            echo Removing "fixed" header files that are actually broken
+            sudo rm -Rf $(find /usr/local/Cellar/gcc@10/ -iname include-fixed)
+            echo Fixing up some binaries to support rpath
+            sudo install_name_tool -add_rpath /usr/local/lib/gcc/10/ -add_rpath @loader_path/. -id @rpath/libgomp.1.dylib /usr/local/lib/gcc/10/libgomp.1.dylib
+            sudo install_name_tool -add_rpath /usr/local/lib/gcc/10/ -add_rpath @loader_path/. -id @rpath/libstdc++.6.dylib /usr/local/lib/gcc/10/libstdc++.6.dylib
+            sudo install_name_tool -add_rpath /usr/local/lib/gcc/10/ -add_rpath @loader_path/. -id @rpath/libgfortran.5.dylib /usr/local/lib/gcc/10/libgfortran.5.dylib
+            sudo install_name_tool -add_rpath /usr/local/lib/gcc/10/ -add_rpath @loader_path/. -id @rpath/libquadmath.0.dylib /usr/local/lib/gcc/10/libquadmath.0.dylib
+            sudo install_name_tool -add_rpath /usr/local/lib/gcc/10/ -add_rpath @loader_path/. -id @rpath/libgcc_s.1.dylib /usr/local/lib/gcc/10/libgcc_s.1.dylib
+            sudo install_name_tool -change $(otool -L /usr/local/lib/gcc/10/libgfortran.5.dylib | grep /usr/local/Cellar/gcc@10/ | cut -f1 -d ' ') @rpath/libquadmath.0.dylib /usr/local/lib/gcc/10/libgfortran.5.dylib
+            sudo install_name_tool -change /usr/local/lib/gcc/10/libgcc_s.1.dylib @rpath/libgcc_s.1.dylib /usr/local/lib/gcc/10/libgomp.1.dylib
+            sudo install_name_tool -change /usr/local/lib/gcc/10/libgcc_s.1.dylib @rpath/libgcc_s.1.dylib /usr/local/lib/gcc/10/libstdc++.6.dylib
+            sudo install_name_tool -change /usr/local/lib/gcc/10/libgcc_s.1.dylib @rpath/libgcc_s.1.dylib /usr/local/lib/gcc/10/libgfortran.5.dylib
+            sudo install_name_tool -change /usr/local/lib/gcc/10/libgcc_s.1.dylib @rpath/libgcc_s.1.dylib /usr/local/lib/gcc/10/libquadmath.0.dylib
+            otool -L /usr/local/lib/gcc/10/*.dylib
+          fi
```

Comment:
Please update all that for GCC 13

Reply:
Done. The "transitive" rpath is already ok for gcc@13, so commands like `install_name_tool -change /usr/local/lib/gcc/10/libgcc_s.1.dylib @rpath/libgcc_s.1.dylib` are not necessary. 
It would be easier if we used gcc@13 (or 14) on macosx-x86_64 too. Is it still necessary to stick with gcc@10 ?

Reply:
> Done. The "transitive" rpath is already ok for gcc@13, so commands like install_name_tool -change /usr/local/lib/gcc/10/libgcc_s.1.dylib @rpath/libgcc_s.1.dylib are not necessary.

I'm pretty sure that's still going to be necessary, let's check the output of `otool -L`

> It would be easier if we used gcc@13 (or 14) on macosx-x86_64 too. Is it still necessary to stick with gcc@10 ?

Sure, would be a good time to upgrade to gcc@13 and fix all broken builds before the next release

Reply:
Do you prefer gcc@13, or the last default homebrew gcc (currently 14) ?

Reply:
> I'm pretty sure that's still going to be necessary, let's check the output of `otool -L`

Looks ok to me

Reply:
Let's not do GCC 14, no